simpler_eda/categorical_eda.py

```diff
@@ -68,12 +67,85 @@ def categorical_plot(
     >>> cars = data.cars()
     >>> categorical_plot(data = cars, 
                         xval = "Origin", 
-                        yval = "count()",
                         color = "Horsepower",
                         title = "Histogram of Origin in Different Levels of Horsepower",
                         plot_height = 100,
                         plot_width = 200
                         )
     """
+    #Checking for valid inputs:
+    if not isinstance(data, pd.DataFrame):
+        raise Exception("the input data has to be a dataframe.")
+    if facet_factor is None and facet_col is not None:
+        raise Exception("facet_factor must be provided along with facet_col.")
+    if facet_factor is not None and facet_col is None:
+        raise Exception("Specify facet_col for facetting the plot")
+    if plot_type not in ["histogram", "density"]:
+        raise Exception("plot_type must be either 'histogram' or 'density'")
+    if opacity <= 0 or opacity >1:
+        raise Exception("opacity must be in range (0, 1)")
+    if xval not in data.columns:
+        raise Exception("xval must be a feature in the input dataframe")
+    if color not in data.columns:
```

Comment:
The default value is None, will there be an issue with this check, I think it will prompt an error message if users do not input a feature? We may consider having the same test for facet_factor as one of the key if-else conditions in the function.

Reply:
Good point!! This has been updated.